src/operator/multicast.ts

```diff
@@ -26,11 +33,35 @@ export function multicast<T>(subjectOrSubjectFactory: Subject<T> | (() => Subjec
       return <Subject<T>>subjectOrSubjectFactory;
     };
   }
-  return new ConnectableObservable(this, subjectFactory);
+
+  if (selector) {
```

Comment:
Here you can create a _single_ `ConnectableObservable` and give it the subjectFactory, then provide that created observable to the `MulticastObservable`:

```
const connectable = new ConnectableObservable(this, subjectFactory);
if (selector) {
  return new MulticastObservable(this, connectable, selector);
} else {
  return connectable;
}
```